common/sf_event.go

```diff
@@ -0,0 +1,55 @@
+package common
+
+import (
+	"encoding/json"
+	"fmt"
+)
+
+type ActionType string
+
+const (
+	AddSF    ActionType = "add"
+	UpdateSF ActionType = "update"
+	DeleteSF ActionType = "delete"
+)
+
+type RequestParameters struct {
+	Action      ActionType
+	NewServices string `json:"newServices,omitempty"`
+	OldServices string `json:"oldServices,omitempty"`
+	NewCustom   string `json:"newCustom,omitempty"`
+	OldCustom   string `json:"oldCustom,omitempty"`
+	NewIsSecret string `json:"newIsSecret,omitempty"`
+	OldIsSecret string `json:"oldIsSecret,omitempty"`
+}
+
+type Detail struct {
+	EventName         string            `json:"eventName"`
+	RequestParameters RequestParameters `json:"requestParameters"`
+}
+
+type SubscriptionFilterEvent struct {
+	Detail Detail `json:"detail"`
```

Comment:
what is the advantage of having this struct instead of using `Detail Detail`

Reply:
I think it's making it more ordered and clear, because without it the `NewSubscriptionFilterEvent` function looks a bit messy:
```go
func NewSubscriptionFilterEvent(params RequestParameters) struct {
	Detail Detail `json:"detail"`
} {
	return struct {
		Detail Detail `json:"detail"`
	}{
		Detail: Detail{
			EventName:         "SubscriptionFilterEvent",
			RequestParameters: params,
		},
	}
}
``` 
what do you think?